packages/astro/test/core-image-layout.test.js

```diff
@@ -0,0 +1,353 @@
+				assert.equal(widths.at(-1), 1600);
+			});
+
+			it('constrained - just has 1x and 2x when smaller than min breakpoint', () => {
+				let $img = $('#local-both img');
+				const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+				assert.deepEqual(widths, [300, 600]);
+			});
+
+			it('fixed - has just 1x and 2x', () => {
+				let $img = $('#local-fixed img');
+				const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+				assert.deepEqual(widths, [800, 1600]);
+			});
+
+			it('full-width: has all breakpoints below image size, ignoring dimensions', () => {
+				let $img = $('#local-full-width img');
+				const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+				assert.deepEqual(widths, [640, 750, 828, 1080, 1280, 1668, 2048]);
+			});
+		});
+
+		describe('remote images', () => {
+			describe('srcset', () => {
+				let $;
+				before(async () => {
+					let res = await fixture.fetch('/remote');
+					let html = await res.text();
+					$ = cheerio.load(html);
+				});
+				it('has srcset', () => {
+					let $img = $('#constrained img');
+					assert.ok($img.attr('srcset'));
+					const srcset = parseSrcset($img.attr('srcset'));
+					const widths = srcset.map((x) => x.w);
+					assert.deepEqual(widths, [640, 750, 800, 828, 1080, 1280, 1600]);
+				});
+
+				it('constrained - has max of 2x requested size', () => {
+					let $img = $('#constrained img');
+					const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+					assert.equal(widths.at(-1), 1600);
+				});
+
+				it('constrained - just has 1x and 2x when smaller than min breakpoint', () => {
+					let $img = $('#small img');
+					const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+					assert.deepEqual(widths, [300, 600]);
+				});
+
+				it('fixed - has just 1x and 2x', () => {
+					let $img = $('#fixed img');
+					const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+					assert.deepEqual(widths, [800, 1600]);
+				});
+
+				it('full-width: has all breakpoints', () => {
+					let $img = $('#full-width img');
+					const widths = parseSrcset($img.attr('srcset')).map((x) => x.w);
+					assert.deepEqual(widths, [640, 750, 828, 1080, 1280, 1668, 2048, 2560]);
+				});
+			});
+		});
+	});
+
+	describe('remote image service', () => {
+		/** @type {import('./test-utils').DevServer} */
+		let devServer;
+		/** @type {Array<{ type: any, level: 'error', message: string; }>} */
+		let logs = [];
+
+		before(async () => {
+			fixture = await loadFixture({
+				root: './fixtures/core-image-layout/',
+				image: {
+					service: testRemoteImageService({ foo: 'bar' }),
+					domains: ['images.unsplash.com'],
+				},
+			});
+
+			devServer = await fixture.startDevServer({
+				logger: new Logger({
+					level: 'error',
+					dest: new Writable({
+						objectMode: true,
+						write(event, _, callback) {
+							logs.push(event);
+							callback();
+						},
+					}),
+				}),
```

Comment:
I don't think we need a logger, since `logs` are never consumed. Same of the other